Feature/ingest comsa

y-chu opened the pull request:
This branch is created off the latest version of main branch.
- Adding ingest_comsa() and related functions to ingest COMSA VA data to RDA in RDAIngest.jl
- test/add_comsa.jl runs ingest_comsa(...) function to add COMSA VA data to database/RDA.sqlite

y-chu committed:
Merge pull request #2 from RDAORG/main

Update branch

y-chu committed:
Adding ingest_comsa and related functions to RDAIngest.jl

kobusherbst reviewed (changes requested):
Yue,

I  think you should consider some re-factoring  There is too much code duplication between the comsa and equivalent champs data ingest functions.  Especially the add_comsa_variables, read_comsa_variables, get_comsa_vocabulary and import_comsa_dataset and the equivalent functions for champs, should be re-factored to place the common code into base functions.

y-chu commented:
> 

Yes many codes are re-used from existing champs functions. I wasn't sure if it'd be preferred to restructure the existing functions - e.g. write some as common functions and setting champs/comsa input source as parameters - so I just added new ones for now. If you're ok with me modifying existing functions then I can go ahead and streamline the coding.

kobusherbst commented:
Yue, absolutely feel free to modify the existing functions.

y-chu committed:
Clean ingest functions

- Created common functions for CHAMPS and COMSA ingest and cleaned related codes
- Add COMSA protocol
- Removed ingest_champs_labtac function and ddd "labtaconly" as an option in ingest_champs()
- Code tested in test/add_comsa.jl
- Reran the functions and ingested CHAMPS and COMSA data to RDA.sqlite

y-chu committed:
REMOVE backup file

y-chu committed:
Update RDAIngest - set source built-in values  as input parameters
